feat(QDOG 154): Create pet diary entry component

JordonH03 opened the pull request:
* Adding `diaryEntry` component and mocks
* Adding list of `diaryEntry` to single pet profile page
* Pet profile page calls pet api to fetch entries for the specific pet 
* Slight reorganization of sub-components

## Screenshots

Desktop:
<img width="1709" height="976" alt="Screenshot 2025-11-04 at 10 38 32 PM" src="https://github.com/user-attachments/assets/c7e1336d-0c07-4c36-a8e6-e1ffee6315f4" />

Mobile:
<img width="438" height="892" alt="Screenshot 2025-11-04 at 10 37 45 PM" src="https://github.com/user-attachments/assets/dd3f39ae-5763-40ff-b929-b0412dc33b1e" />

JordonH03 committed:
chore: Reorganize passwordReq and sidebar components

JordonH03 committed:
chore: Reorganize passwordReq and sidebar components

JordonH03 committed:
chore: format and lint fixes to pet pages

JordonH03 committed:
chore: format and lint fixes to pet pages

JordonH03 committed:
feat: Add pet diary entries to pet profile page

JordonH03 committed:
feat: Add pet diary entries to pet profile page

JordonH03 committed:
fix: sidebar test import path

JordonH03 committed:
fix: sidebar test import path

nataniella-o commented:
nit for later: the edit button & "view all" are missing.
aside from that everything looks great!

arasantos committed:
chore: Update ci for regression tests (#43)

* chore: Update ci for regression tests

Do selective regression tests for ci when pull requests are made to main and develop
Run authentication, create diary and pet, and integration tests as those seem to be high traffic or high risk

* chore: fix typo in folder name

* Run all integration tests

nataniella-o committed:
feature(QDOG 138): create pet diary dashboard (#42)

* feat(QDOG-138): create pet diary dashboard layout

* feat(QDOG-138): fix failed (some) tests from create new diary entry cause of dashboard changes

* feat(QDOG-138): revert pet diary create page

* feat(QDOG-138): update pet diary create page to have preselected noteType

* feat(QDOG-138): update & fix pet diary create entry tests

* feat(QDOG-138): create pet diary dashboard page test

* feat(QDOG-138): fix build issue

* feat(QDOG-138): fix build issue

* feat(QDOG-138): aesthetic changes- svg icons & color variables in diary dashboard module

* feat(QDOG-138): address nit's from pr comments

* feat(QDOG-138): fix tab in diary dashboard styling

* feat(QDOG-138): adding responsive styling

* feat(QDOG-138): minor fix to responsive styling

* Remove .gitattributes

* feat(QDOG-138): add AI acknowledgement to some files

* feat(QDOG-138): edit Quick Add buttons to use noteTypeOptions, exclude Other

* feat(QDOG-138): add line to trigger prettier

* re-added git attributes

---------

Co-authored-by: Nataniella Ogogo <nataniellaogogo@wifi-wpa-cw2-140-193-88-169.wifi.umanitoba.ca>
Co-authored-by: basmaabdulrazaq <abdulrab@myumanitoba.ca>

arasantos commented on src/app/owner/pets/[id]/page.test.tsx:
```diff
@@ -24,66 +24,106 @@ jest.mock("~util/strings/format-pet", () => {
-    it("renders error component on API error", async () => {
-        jest.spyOn(PetsAPI, "getPet").mockRejectedValue("API Error");
-        await render(<PetProfilePage />);
+    describe("Diary entries", () => {
```
nit: for this group, seems like the `mockGetPet.mockResolvedValue(mockPets[0]);` and `await render(<PetProfilePage />);` are getting used in all the tests under this group. Maybe we can put that in a beforeEach() ?

JordonH03 replied:
I can extract mockGetPet into a beforeEach, but since we're testing the diary entries, the mocks must be called before rendering, so each test needs an explicit call to render

arasantos replied:
Oh okay I see!

arasantos commented on src/app/owner/pets/[id]/page.tsx:
```diff
@@ -33,22 +37,32 @@ export default function PetProfilePage() {
                 <Button
                     component={Link}
-                    href='/owner/dashboard'
+                    href='/owner/pets/dashboard'
```
nit: the back button padding (or space) from the left is not the same as the menu's space/padding from the right. Can we make that the same so they're balanced? 

I'm referring to the mobile image in pr description.

arasantos commented on src/app/owner/pets/dashboard/page.module.scss:
```diff
@@ -43,105 +43,104 @@
-  font-size: 30px;
-//   opacity: 0.5;
+    font-size: 30px;
+    //   opacity: 0.5;
```
We can probably remove this commented code, I know it's not originally from this pr but I just noticed it now

arasantos reviewed (changes requested):
Looks really good! I left some comments, most are like nitpicks

BasmaAbdulRazaq reviewed (approved):
looks good! just resolve the merge conflicts

JordonH03 committed:
chore: Update styles

JordonH03 committed:
chore: Update styles

JordonH03 committed:
Merge branch 'feat/QDOG-154-diary-entry-component' of github.com:4350-fall-2025/frontend-1 into feat/QDOG-154-diary-entry-component

arasantos reviewed (approved):
looks good! thanks for making the changes
